### PR TITLE
White phosphorus grenades .

### PR DESCRIPTION
**Pull request by thedarkcrusader**
Adds Parts to of WP grenades  anew icend grenade for deadly  applications

## About The Pull Request
This PR adds in the ever-popular WIlly pete  Serbian Arms? Technomancer???  White phosphorus grenade A grenade which is supposed to be terrifyingly efficient at burning people and Area Denil   the WP grenade detonates like a normal smoke grenade but with a catch the WP grenade will send out embers when it detonates these embers on hit with any object or mob will deploy another WP cloud.
image

## Why It's Good for the Game
New Grenades are allways good especially white Phosphorus 

## Changelog
🆑 :Theblackflag
add: Willy Pete grenade
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 1]
```diff
@@ -71,3 +71,38 @@
...
+	playsound(loc, 'sound/effects/smoke.ogg', 50, 1, -3)
+	smoke.set_up(5, 0, usr.loc)
+	spawn(0)
+		smoke.start()
+		sleep(10)
+		smoke.start()
+		sleep(10)
+		smoke.start()
+		sleep(10)
+		smoke.start()
```
Comment (Pink-Chink) [inline code/game/objects/items/weapons/grenades/frag.dm:98]:
```suggestion
	smoke.set_up(5, 0, get_turf(loc))
	smoke.start()
	..()
```
It produces a good amount of smoke and the fragments help spread it around. If it's too weak, you can always buff the smoke or increase the number of embers.

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 2]
```diff
@@ -71,3 +71,38 @@
...
+
+/obj/item/grenade/frag/white_phosphorous/New()
+	..()
+	smoke = new
+	smoke.attach(src)
+
+/obj/item/grenade/frag/white_phosphorous/Destroy()
+	qdel(smoke)
+	smoke = null
+	return ..()
```
Comment (Pink-Chink) [inline code/game/objects/items/weapons/grenades/frag.dm:108]:
```suggestion
	return ..()

```
End line is needed at the end of the file.

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 3]
```diff
@@ -71,3 +71,38 @@
 	blast_radius = 0
 	damage_step = 8
 	spread_range = 7
+
+/obj/item/grenade/frag/white_phosphorous
+	name = "SA WPG \"Geneva\""
+	desc = "A modernized Incendiary hailing popular use within technomancer tribes and Mercnary assualt teams. use this on unruly crowds."
+	icon_state = "white_pho"
```
Comment (Pink-Chink) [inline code/game/objects/items/weapons/grenades/frag.dm:78]:
```suggestion
	icon_state = "white_phos"
```

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 4]
```diff
@@ -71,3 +71,38 @@
...
+	item_state = "fraggrenade"
+	fragment_type = /obj/item/projectile/ember
```
Comment (Pink-Chink) [inline code/game/objects/items/weapons/grenades/frag.dm:80]:
```suggestion
	fragment_type = /obj/item/projectile/bullet/pellet/fragment/ember
```
After you move the ember to fragments.dm.

File: code/modules/projectiles/projectile/special.dm [hunk 5]
```diff
@@ -254,5 +254,4 @@
 	src.visible_message(SPAN_WARNING("\The [src] explodes in a bright light!"))
 	new /obj/effect/decal/cleanable/ash(src.loc)
 	playsound(src, 'sound/effects/flare.ogg', 100, 1)
-	new /obj/effect/effect/smoke/illumination(T, brightness=max(flash_range*3, brightness), lifetime=light_duration, color=COLOR_RED)
-
+	new /obj/effect/effect/smoke/illumination(T, brightness=max(flash_range*3, brightness), lifetime=light_duration, color=COLOR_RED)
```
Comment (Pink-Chink) [inline code/modules/projectiles/projectile/special.dm:257]:
```suggestion
	new /obj/effect/effect/smoke/illumination(T, brightness=max(flash_range*3, brightness), lifetime=light_duration, color=COLOR_RED)

```

File: code/modules/projectiles/projectile/fragment.dm [hunk 6]
```diff
@@ -38,3 +38,18 @@
...
+    check_armour = ARMOR_ENERGY
+    embed = 0
+    pellets = 1
+    range_step = 5
+
+/obj/item/projectile/bullet/pellet/fragment/ember/on_hit(atom/target)
+    var/datum/effect/effect/system/smoke_spread/white_phosphorous/S = new /datum/effect/effect/system/smoke_spread/white_phosphorous
+    S.set_up(1, 0, get_turf(src))
+    S.start()
+    return TRUE
```
Comment (Pink-Chink) [inline code/modules/projectiles/projectile/fragment.dm:55]:
```suggestion
    return TRUE

```

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 7]
```diff
@@ -71,3 +71,33 @@
...
+	num_fragments = 10
+	fragment_damage = 5
+	blast_radius = 0
+	damage_step = 5
+	spread_range = 7
+
+/obj/item/grenade/frag/white_phosphorous/prime()
+	playsound(loc, 'sound/effects/smoke.ogg', 50, 1, -3)
+	smoke.set_up(5, 0, usr.loc)
+	smoke.set_up(5, 0, get_turf(loc))
```
Comment (Pink-Chink) [inline code/game/objects/items/weapons/grenades/frag.dm:91]:
```suggestion
	smoke.set_up(5, 0, get_turf(loc))
```
Oops.

Comment (Pink-Chink) [inline code/game/objects/items/weapons/grenades/frag.dm:91]:
```suggestion
	smoke.set_up(5, 0, get_turf(loc))
```

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 8]
```diff
@@ -71,3 +71,32 @@
 	blast_radius = 0
 	damage_step = 8
 	spread_range = 7
+
+/obj/item/grenade/frag/white_phosphorous
+	name = "SA WPG \"Geneva\""
+	desc = "A modernized Incendiary hailing popular use within technomancer tribes and Mercnary assualt teams. use this on unruly crowds."
```
Comment (Firefox13) [inline code/game/objects/items/weapons/grenades/frag.dm:77]: Can we ease up on "HAHA GENEVA CONVENTION'" jokes?
```suggestion
	desc = "A modernized incendiary hailing popular use within shock troopers and other assault troops. Use with care, highly flammable."
```

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 9]
```diff
@@ -71,3 +71,32 @@
 	blast_radius = 0
 	damage_step = 8
 	spread_range = 7
+
+/obj/item/grenade/frag/white_phosphorous
+	name = "SA WPG \"Geneva\""
```
Comment (Firefox13) [inline code/game/objects/items/weapons/grenades/frag.dm:76]:
```suggestion
	name = "SA WPG \"Basel\""
```
Named after a different Switzerland city

Comment (SirRichardFrancis) [reply code/game/objects/items/weapons/grenades/frag.dm:76]: Unbaseled and soulless.

Comment (Humonitarian) [reply code/game/objects/items/weapons/grenades/frag.dm:76]: Only oldNT ones should be called by Swiss names, I'd suggest using a serbian name for this one

File: code/modules/projectiles/projectile/special.dm [hunk 10]
```diff
@@ -255,4 +255,3 @@
 	new /obj/effect/decal/cleanable/ash(src.loc)
 	playsound(src, 'sound/effects/flare.ogg', 100, 1)
 	new /obj/effect/effect/smoke/illumination(T, brightness=max(flash_range*3, brightness), lifetime=light_duration, color=COLOR_RED)
-
```
Comment (Humonitarian) [inline code/modules/projectiles/projectile/special.dm:258]: Lost the newline?

File: code/game/objects/items/weapons/grenades/frag.dm [hunk 11]
```diff
@@ -71,3 +71,33 @@
...
+	desc = "A modernized incendiary hailing popular use within assault troops of all kinds. Use with care, highly flammable."
+	icon_state = "white_phos"
+	item_state = "fraggrenade"
+	fragment_type = /obj/item/projectile/bullet/pellet/fragment/ember
+	var/datum/effect/effect/system/smoke_spread/white_phosphorous/smoke
+	num_fragments = 10
+	fragment_damage = 5
+	blast_radius = 0
+	damage_step = 5
+	spread_range = 7
```
Comment (Pink-Chink) [inline code/game/objects/items/weapons/grenades/frag.dm:86]:
```suggestion
	num_fragments = 10
	fragment_damage = 5
	blast_radius = 0
	damage_step = 5
	spread_range = 7
	var/datum/effect/effect/system/smoke_spread/white_phosphorous/smoke
```

**Review by Pink-Chink — APPROVED** [review]
Dev del approved.

**Comment by Firefox13** [comment]
That mount of WP grenades looks suspicious...

**Pull request merged**